### PR TITLE
[SYCLomatic] Support -isystem in makefile generator

### PR DESCRIPTION
**Pull request by yihwang-nv**
Signed-off-by: Wang, Yihan <yihan.wang@intel.com>

### CODE REVIEW

File: clang/lib/DPCT/GenMakefile.cpp [hunk 1]
```diff
@@ -185,6 +213,9 @@ static void getCompileInfo(
         NewOptions += OutDirectory.c_str();
         NewOptions += " ";
 
+      } else if (llvm::StringRef(Option).startswith("-isystem")) {
+        ISISystem = true;
+        std::cout << "I have found an -isystem\n";
```
Comment (tomflinda) [inline clang/lib/DPCT/GenMakefile.cpp]: Line 218 should be removed。

Comment (yihwang-nv) [reply clang/lib/DPCT/GenMakefile.cpp]: done

**Review by tomflinda — APPROVED** [review]
LGTM

File: clang/lib/DPCT/GenMakefile.cpp [hunk 2]
```diff
@@ -153,9 +153,37 @@ static void getCompileInfo(
...
     const std::string Directory = Entry.second[0];
     std::unordered_set<std::string> DuplicateDuplicateFilter;
     for (const auto &Option : Entry.second) {
+
+      if (ISISystem) {
+        ISISystem = false;
+        std::string IncPath = Option;
+
+        if (!llvm::sys::fs::exists(IncPath)) {
+          // Skip including path that does not exist.
```
Comment (zhimingwang36) [inline clang/lib/DPCT/GenMakefile.cpp]: for user, migration machine may not be the build machine,  so maybe it's not good to remove the non exist inlucde in migration machine.

Comment (yihwang-nv) [reply clang/lib/DPCT/GenMakefile.cpp]: OK

Comment (yihwang-nv) [reply clang/lib/DPCT/GenMakefile.cpp]: done

**Review by ziranzha — APPROVED** [review]

**Pull request merged**